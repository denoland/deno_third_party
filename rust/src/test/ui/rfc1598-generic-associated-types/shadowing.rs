#![feature(generic_associated_types)]

//FIXME(#44265): The lifetime shadowing and type parameter shadowing
// should cause an error. Now it compiles (erroneously) and this will be addressed
// by a future PR. Then remove the following:
// compile-pass

trait Shadow<'a> {
    type Bar<'a>; // Error: shadowed lifetime
}

trait NoShadow<'a> {
    type Bar<'b>; // OK
}

impl<'a> NoShadow<'a> for &'a u32 {
    type Bar<'a> = i32; // Error: shadowed lifetime
}

trait ShadowT<T> {
    type Bar<T>; // Error: shadowed type parameter
}

trait NoShadowT<T> {
    type Bar<U>; // OK
}

impl<T> NoShadowT<T> for Option<T> {
    type Bar<T> = i32; // Error: shadowed type parameter
}

fn main() {}
