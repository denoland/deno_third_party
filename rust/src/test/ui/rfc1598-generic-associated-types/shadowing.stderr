warning: the feature `generic_associated_types` is incomplete and may cause the compiler to crash
  --> $DIR/shadowing.rs:1:12
   |
LL | #![feature(generic_associated_types)]
   |            ^^^^^^^^^^^^^^^^^^^^^^^^

